src/Bicep.Core/TypeSystem/TypeHelper.cs

```diff
@@ -688,37 +687,42 @@ private static ObjectType TransformProperties(ObjectType input, Func<NamedTypePr
                 input.Name,
                 input.ValidationFlags,
                 input.Properties.Values.Select(transformFunc),
-                input.AdditionalProperties,
+                input.AdditionalProperties is not null ? input.AdditionalProperties with { } : null,
```

Comment:
Why do we need this?

Reply:
Isn't `input.AdditionalProperties` a `TypeProperty?` (an immutable record)?

Reply:
Oh my mistake. I forgot that constructor only properties are init only.

Reply:
I was working with record structs recently and mistaken registered the lack of "readonly" on it as mutable 🤦‍♂️